File: 44翻转单词序列.java
// 最直观的方法
public class Solution {
    public String ReverseSentence(String str) {
        if(str == null || str.length() == 0) return "";
        if(str.trim().equals("")) return str;
        
        String[] words = str.split(" ");
        StringBuilder sb = new StringBuilder();
        for(int i = words.length-1; i >= 0; i--) {
            sb.append(words[i]);
            if(i > 0) {
                sb.append(" ");
            }
        }
        return sb.toString();
    }
}

File: 44翻转单词序列.py
# -*- coding:utf-8 -*-
class Solution:
    def ReverseSentence(self, s):
        # write code here
        if s is None: return None
        
        str_list = s.split(' ')
        if len(str_list) < 2:
            return s
        else:
            return ' '.join(str_list[::-1])